esx_service/utils/vmdk_utils.py

```diff
@@ -40,6 +40,9 @@
 # glob expression to match end of 'delta' (aka snapshots) file names.
 SNAP_SUFFIX_GLOB = "-[0-9][0-9][0-9][0-9][0-9][0-9].vmdk"
 
+# regexp for finding datastore path "[datastore] path/to/file.vmdk" from full vmdk path
+DATASTORE_PATH_REGEXP = r"^/vmfs/volumes/([a-zA-Z0-9_-]+?)/(.*)"
```

Comment:
Datastores can have pretty much anything in the name. It's a symlink after all. I suspect there are interesting corner cases wich will be broken by this regexp. 
Also, I think this may NOT work with the tenant since <tenant> componet of the path is not limited to  `[a-zA-Z0-9_-]` (aka `w`) @lipingxue 

maybe something like this `^/vmfs/volumes/([^/]+)/(.*\.vmdk)$` ?